pure.zsh

```diff
@@ -390,6 +390,10 @@ prompt_pure_async_callback() {
 	local do_render=0
 
 	case $job in
+		\[async])
```

Comment:
I think we need a way to differentiate between the two types of `[async]` errors that can be received here. One is https://github.com/mafredri/zsh-async/blob/98d32afcce8e328be86f381c529e951943e57d19/async.zsh#L288 which returns an exit code of 1 (albeit that behavior is undefined). And then there's the new error (https://github.com/mafredri/zsh-async/blob/98d32afcce8e328be86f381c529e951943e57d19/async.zsh#L320) that indicates the worker has been killed. Just so happens it returns an exit code of 2 now, also undefined.

We could rename the new error to `[async/crash]`. Or we could start defining the error codes. For now checking for `2` should be sufficient though.